src/molecule_docker/test/test_func.py

```diff
@@ -68,3 +68,11 @@ def test_dockerfile_with_context() -> None:
         cmd = ["molecule", "--debug", "test"]
         result = run_command(cmd)
         assert result.returncode == 0
+
+
+def test_env_substitution() -> None:
```

Comment:
Could you define the environment variable, `MOLECULE_ROLE_IMAGE ` in test case?

The environment is defined actually, and root cause is the `molecule.yml` is overwritten by the `Synchronization the context` task.